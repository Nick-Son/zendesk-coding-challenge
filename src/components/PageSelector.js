import React from 'react'

const PageSelector = (props) => (
  <div className="page-selector">
    <ul className="page-selector__list">
      { props.pageNumbers &&
        props.pageNumbers.map(page => (
          (
            <li 
              className="page-selector__link"
              key={page} 
              id={page} 
              onClick={() => props.onPageChange(page)}
              >{page + 1}
            </li>
          )
        ))
      }
    </ul>
  </div>
)

export default PageSelector