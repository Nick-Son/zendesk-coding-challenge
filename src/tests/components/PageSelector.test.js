import React from 'react';
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJSON from 'enzyme-to-json'
import PageSelector from '../../components/PageSelector';

Enzyme.configure({
  adapter: new Adapter()
})

test('Should render PageSelector correctly with props', () => {
  const pageNumbers = [0, 1, 2, 3, 4]
  const wrapper = shallow(<PageSelector currentPage={0} pageNumbers={pageNumbers} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should change current page on click', () => {
  const pageNumbers = [0, 1, 2, 3, 4]
  const onPageChangeSpy = jest.fn()
  const wrapper = shallow(<PageSelector currentPage={0} pageNumbers={pageNumbers} onPageChange={onPageChangeSpy} />)
  wrapper.find('li').at(1).simulate('click')
  expect(onPageChangeSpy).toHaveBeenCalledWith(1)
})
