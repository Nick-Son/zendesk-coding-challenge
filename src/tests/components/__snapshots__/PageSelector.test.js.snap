// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PageSelector correctly with props 1`] = `
<ul
  className="page-selector"
>
  <li
    className="current-page page-selector__link"
    id={0}
    key="0"
    onClick={[Function]}
  >
    1
  </li>
  <li
    className="page-selector__link"
    id={1}
    key="1"
    onClick={[Function]}
  >
    2
  </li>
  <li
    className="page-selector__link"
    id={2}
    key="2"
    onClick={[Function]}
  >
    3
  </li>
  <li
    className="page-selector__link"
    id={3}
    key="3"
    onClick={[Function]}
  >
    4
  </li>
  <li
    className="page-selector__link"
    id={4}
    key="4"
    onClick={[Function]}
  >
    5
  </li>
</ul>
`;
